File: lib/src/domain/enums/expense/expense_category_enum.dart
enum ExpenseCategoryEnum {
  food,
  debt,
  health,
  unknown,
  housing,
  shopping,
  transport,
  entertainment;

  static ExpenseCategoryEnum fromString(String? value) => switch (value) {
    'food' => .food,
    'debt' => .debt,
    'health' => .health,
    'housing' => .housing,
    'shopping' => .shopping,
    'transport' => .transport,
    'entertainment' => .entertainment,
    _ => .unknown,
  };
}

File: app/Expense/Domain/Enums/ExpenseCategoryEnum.php
<?php

declare(strict_types=1);

namespace App\Expense\Domain\Enums;

enum ExpenseCategoryEnum: string
{
    case Food = 'food';
    case Health = 'health';
    case Housing = 'housing';
    case Leisure = 'leisure';
    case Shopping = 'shopping';
    case Services = 'services';
    case Transport = 'transport';
    case Education = 'education';
    case Subscriptions = 'subscriptions';
    case Other = 'other';
}
